libs/utils/src/logging.rs

```diff
@@ -178,6 +202,29 @@ pub fn init(
     Ok(())
 }
 
+pub fn init_otel_tracing(otel_enablement: OtelEnablement) -> anyhow::Result<Option<OtelGuard>> {
```

Comment:
This could use a comment:

- What's the default configuration?
- What do you do with the returned guard? It includes a 'dispatch', what should you do with that? What happens when you drop the guard?
- Per discussion, this does not hook into the default tracing span hierarchy, right? Please point that out explicitly.

Should this function be in the `libs/tracing-utils` crate instead? Per discussion, this is separate from logging and the normal tracing hierarchy. Or in 'pageserver/', since that's the only component using this.

Reply:
Good points. Moved it into `tracing-utils`. Other services may use it in the future (e.g. safekeepers). Also included a doc comment.